### ISSUE TITLE
flake8-logging-format should check `.log(level, msg)` calls too

### ISSUE THREAD

**Issue by andersk** [charliermarsh/ruff#2562]
```python
import logging
logger = logging.getLogger("test")
a, b = "a", "b"
logger.error(f"{a} {b}")  # raises G004 Logging statement uses f-string
logger.log(logging.ERROR, f"{a} {b}")  # should raise G004, but doesn’t
```

Also reported upstream at globality-corp/flake8-logging-format#69.

**Comment by charliermarsh** [comment]
Can do!

**Comment by dhruvmanila** [comment]
So, I'm looking into this and have a few queries:

As per the [documentation of the `log` function](https://docs.python.org/3/library/logging.html#logging.Logger.log), it says that the `level` argument could be any integer level. In addition to that, it allows [adding](https://docs.python.org/3/library/logging.html#logging.addLevelName) of custom log levels as well. The user could pass in any arbitrary integer as well and that works.

_Should we support any arbitrary integer levels or only the ones defined in the standard library?_

This is more towards the implementation side: I see that the implementation has been grouped together instead of each rule being a separate module. With the support of `logging.log`, special handling would be required where it's either one or the other. Are there any plans to refactor into separate modules or is this ok for now?

**Comment by andersk** [comment]
We have no need to validate or even inspect the `level` argument. (It’s probably going to be a variable anyway.) We just need to apply the same string formatting rules to `logger.log(level, msg)` that we would apply to `logger.debug(msg)`.
